crates/bevy_core/src/time/time.rs

```diff
@@ -34,58 +49,171 @@ impl Time {
     pub(crate) fn update_with_instant(&mut self, instant: Instant) {
         if let Some(last_update) = self.last_update {
             self.delta = instant - last_update;
-            self.delta_seconds_f64 = self.delta.as_secs_f64();
             self.delta_seconds = self.delta.as_secs_f32();
+            self.delta_seconds_f64 = self.delta.as_secs_f64();
+            self.time_since_startup += self.delta.mul_f32(self.relative_speed);
+        } else {
+            self.first_update = Some(instant);
+            self.time_since_startup += (instant - self.startup).mul_f32(self.relative_speed);
         }
 
-        let duration_since_startup = instant - self.startup;
-        self.seconds_since_startup = duration_since_startup.as_secs_f64();
         self.last_update = Some(instant);
     }
 
-    /// The delta between the current tick and last tick as a [`Duration`]
+    /// The delta between the current and last update, as a [`Duration`].
     #[inline]
     pub fn delta(&self) -> Duration {
         self.delta
     }
 
-    /// The delta between the current and last tick as [`f32`] seconds
+    /// The delta between the current and last update, as [`f32`] seconds
     #[inline]
     pub fn delta_seconds(&self) -> f32 {
         self.delta_seconds
     }
 
-    /// The delta between the current and last tick as [`f64`] seconds
+    /// The delta between the current and last update, as [`f64`] seconds.
     #[inline]
     pub fn delta_seconds_f64(&self) -> f64 {
         self.delta_seconds_f64
     }
 
-    /// The time since startup in seconds
+    /// The *virtual* delta between each fixed update step, as a [`Duration`].
+    /// Note that the true delta between fixed update steps varies.
     #[inline]
-    pub fn seconds_since_startup(&self) -> f64 {
-        self.seconds_since_startup
+    pub fn fixed_delta(&self) -> Duration {
+        self.fixed_delta
     }
 
-    /// The [`Instant`] the app was started
+    /// The *virtual* delta between each fixed update step, as [`f32`] seconds.
+    /// Note that the true delta between fixed update steps varies.
     #[inline]
-    pub fn startup(&self) -> Instant {
-        self.startup
+    pub fn fixed_delta_seconds(&self) -> f32 {
+        self.fixed_delta_seconds
+    }
+
+    /// The *virtual* delta between each fixed update step, as [`f64`] seconds.
+    /// Note that the true delta between fixed update steps varies.
+    #[inline]
+    pub fn fixed_delta_seconds_f64(&self) -> f64 {
+        self.fixed_delta_seconds_f64
+    }
+
+    /// Changes [`Time::fixed_delta`]. Users are strongly cautioned against modifying
+    /// the step rate this way, except during startup.
+    /// Prefer [`Time::set_relative_speed`] to ensure that consumers of
+    /// [`Time::fixed_delta`] produce consistent behavior.
```

Comment:
Actually, whether changing this or `relative_speed`, that will destabilize a fixed-time simulation. So the caution should be on both.

Reply:
You're incorrect. Changing the relative speed does not change the value of `Time::fixed_delta`, which systems should be using for consistency. Only the number of times those systems run would change.

Reply:
Right, I had misunderstood how the accumulator worked. You're right, `fixed_delta` is not touched so this does not affect the stability of the fixed-step simulation itself. However, this simulation, or its inputs/outputs, will necessarily at some point be used in conjunction with other data computed during the variable-step (game) simulation; for example results of the physics engine will move bodies, which might trigger animations or other effect in the gameplay logic. When that happen, and `relative_speed` changes, that can cause some instability. So I think we should caution the users anyway about time scaling.

Reply:
> However, this simulation, or its inputs/outputs, will necessarily at some point be used in conjunction with other data computed during the variable-step (game) simulation; for example results of the physics engine will move bodies, which might trigger animations or other effect in the gameplay logic. When that happen, and relative_speed changes, that can cause some instability.

There won't be instability. They just need to be synchronized to the same clock, i.e. those animations need to be affected by `relative_speed` too. So I need have `delta()` affected by `relative_speed` to complement `time_since_startup()` and add a `real_delta()` to complement `real_time_since_startup()`.

Reply:
Yeah I still think there's potential for instability from those syncs, but that's probably minor and inconsequential, so can be ignored.

Reply:
By synchronized, I just meant that entities and their animations should run on the same clock (either affected by `relative_speed` or not). There's no "sync" action.

And when using a fixed timestep, you would interpolate to render the correct animation state.

I don't see any numerical instabilities, only the potential for user error.